discovery/syncer.go

```diff
@@ -715,17 +731,12 @@ func (g *GossipSyncer) processChanRangeReply(msg *lnwire.ReplyChannelRange) erro
 
 		// If we've previously received a reply for this query, look at
 		// its last block to ensure the current reply properly follows
-		// it.
+		// it. The current reply must start from the next block of
+		// previous reply's last block.
 		if g.prevReplyChannelRange != nil {
 			prevReply := g.prevReplyChannelRange
 			prevReplyLastHeight := prevReply.LastBlockHeight()
-
-			// The current reply can either start from the previous
-			// reply's last block, if there are still more channels
-			// for the same block, or the block after.
-			if msg.FirstBlockHeight != prevReplyLastHeight &&
-				msg.FirstBlockHeight != prevReplyLastHeight+1 {
-
+			if msg.FirstBlockHeight != prevReplyLastHeight+1 {
```

Comment:
will this be incompatible with peers running older lnd versions?

Reply:
Good point, reverted to maintain compat with older versions.